src/core/account_builder.cpp

```diff
@@ -0,0 +1,278 @@
+/* 
+File: account_builder.cpp
+
+Brief:  Account builder performs exception handling and guides the user
+        through the process of creating an account.
+
+Authors: Daniel Mendez 
+         Alex Salazar
+         Arman Alauizadeh 
+         Alexander DuPree
+         Kyle Zalewski
+         Dominique Moore
+
+https://github.com/AlexanderJDupree/ChocAn
+ 
+*/
+
+#include <ChocAn/core/utils/account_builder.hpp>
+#include <ChocAn/core/utils/validators.hpp>
+#include <algorithm>
+
+
+bool Account_Builder::buildable() const
+{
+    return build_phase.empty() && account_accepted;
+}
+
+Account Account_Builder::build()
+{
+    invalid_account_build errors("",{});
+
+    if (!buildable())
+        throw invalid_account_build(errors.default_msg, {errors.building_unbuildable});
+
+    if (account_field.at(current_type) == "Manager")
+        return Account(name.value(), address.value(), Manager(), id_generator);
+
+    if (account_field.at(current_type) == "Provider")
+        return Account(name.value(), address.value(), Provider(), id_generator);
+    
+    return Account(name.value(), address.value(), Member(), id_generator);
+
+}
+
+Account_Builder &Account_Builder::reset()
+{
+
+    while (!build_phase.empty())
+        build_phase.pop();
+
+    Build_Instructions building_phase;
+    
+    build_phase.push(building_phase.street_address);
+    build_phase.push(building_phase.full_name);
+    build_phase.push(building_phase.account_type);
+
+    name.reset();
+    address.reset();
+    zip.reset();
+    issues.reset();
+
+    account_accepted = false;
+
+    return *this;
+}
+
+void Account_Builder::set_current_field(const std::string &input)
+{
+
+    Build_Instructions building_phase;
+   
+    if (build_phase.empty())
+    {
+
+        account_accepted = confirm_account(input);
```

Comment:
Right now, if during the confirm account phase if hit enter or type garbage I have to restart the entire account creation process. I would like to loop until I enter either a yes or no.

Reply:
going to be fixed during build phase refactor. will not have a confirm_account method.